assets/js/sidebar.js

```diff
@@ -0,0 +1,53 @@
+var $ = require('jquery')
+
+var body = $('body')
+var sidebar = $('.sidebar')
+
+var bodyClass = 'sidebar-open'
+var duration = 300
+var width = '300px'
+
+function closeSidebar () {
+  sidebar.animate({
+    'flex-basis': 0
+  }, duration, function () {
+    body.removeClass(bodyClass)
+  })
+}
+
+function openSidebar (immediate) {
+  var dur = duration
+  if (immediate) {
+    dur = 0
+  }
+  sidebar.animate({
+    '-webkit-flex-basis': width,
+    '-moz-flex-basis': width,
+    '-ms-flex-basis': width,
+    'flex-basis': width
+  }, 0, function () {
+    body.addClass(bodyClass)
+  })
+}
+
+function toggleSidebar () {
+  if (sidebar.width() > 0) {
+    closeSidebar()
+  } else {
+    openSidebar()
+  }
+}
+
+function init () {
+  $('.sidebar-toggle').click(function () {
+    toggleSidebar()
+  })
+
+  if ($(window).width() > 768) {
```

Comment:
IMO, this kind of detection is more related with styling stuff... So, why not to use a *css media query* in this case?